### PR TITLE
add check:hoisted command to theia cli

### PR DESCRIPTION
**Pull request by akosyakov**
in order to:
- get rid of usages of ts-node
- allow to check final apps

### CODE REVIEW

File: dev-packages/cli/src/check-hoisting.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+ * This script makes sure all the dependencies are hoisted into the root `node_modules` after running `yarn`.
+ *  - https://github.com/theia-ide/theia/pull/2994#issuecomment-425447650
+ *  - https://github.com/theia-ide/theia/pull/2994#issuecomment-425649817
+ *
+ * If you do not want to bail the execution, set the `THEIA_CHECK_HOISTING_NO_BAIL` environment variable to `true`. Use `cross-env`.
+ * ```json
+ *   ...
+ *   "scripts": {
+ *     ...,
+ *     "prepare:hoisting": "cross-env THEIA_CHECK_HOISTING_NO_BAIL=true node scripts/check-hoisting",
```
Comment (kittaakos) [inline dev-packages/cli/src/check-hoisting.ts]: Please keep this line in sync with https://github.com/theia-ide/theia/pull/4138/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R51.

Also, maybe we could remove `cross-env` from the main `package.json` and move it to either the `@theia/application-manager` or the `@theia/cli`. Thoughts?

Comment (akosyakov) [reply dev-packages/cli/src/check-hoisting.ts]: @kittaakos i was thinking to replace it with cli option, WDYT? something like `theia check:hoisted --silent` then we don't need `cross-env`

Comment (kittaakos) [reply dev-packages/cli/src/check-hoisting.ts]: Sure, your approach is even better.

Comment (akosyakov) [reply dev-packages/cli/src/check-hoisting.ts]: @kittaakos I've went with `suppress` option, no `silent`. Good to merge like that?

**Comment by kittaakos** [comment]
I have tried it with e46d5fc0, it worked as expected.

I guess we cannot run the hosting check upfront, just like we do it on the `master` branch, right?

**Comment by akosyakov** [comment]
> I guess we cannot run the hosting check upfront, just like we do it on the master branch, right?

>es, we need to compile it first. It is fine on local install, since you see it as the last. on travis maybe we should extract it to a separate script that it does not get lost between build and test output.

**Pull request merged**